src/cacheUntilType.js

```diff
@@ -0,0 +1,51 @@
+import { OuterSubscriber } from 'rxjs/OuterSubscriber';
+import { subscribeToResult } from 'rxjs/util/subscribeToResult';
+
+
+export function cacheUntilType(type) {
+  return this.lift(new CacheUntilTypeOperator(type));
+}
+
+class CacheUntilTypeOperator {
```

Comment:
don't do anything yet, but if we do go this route I'd rather use closures instead of classes so that we have a smaller bundle size since this is should not be a hot code path.